velox/dwio/common/DataBuffer.h

```diff
@@ -141,8 +141,12 @@ class DataBuffer {
     append(offset, src.data() + srcOffset, items);
   }
 
-  void append(uint64_t offset, const T* FOLLY_NONNULL src, uint64_t items) {
-    reserve(offset + items);
+  void append(
```

Comment:
I don't think we should make this low level object too smart like this.  Just add a `reserve` call to `parquet::DataBufferSink::Write` should fix your use case

Reply:
Sure. I have added the `reserve `call in `parquet::DataBufferSink::Write`. Please help to review again. Thanks.